src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt

```diff
@@ -57,6 +57,24 @@ class PrisonerSearchService(
     } ?: Result.NoMatch
   }
 
+  private fun queryBy(searchCriteria: PrisonerListCriteria, queryBuilder: (searchCriteria: PrisonerListCriteria) -> BoolQueryBuilder?): Result {
+    val query = queryBuilder(searchCriteria)
+    return query?.let {
+      val searchSourceBuilder = SearchSourceBuilder().apply {
+        query(query)
+      }
+      val searchRequest = SearchRequest(arrayOf(getIndex()), searchSourceBuilder)
+      val prisonerMatches = getSearchResult(searchClient.search(searchRequest))
+      return if (prisonerMatches.isEmpty()) Result.NoMatch else Result.Match(prisonerMatches)
+    } ?: Result.NoMatch
+  }
+
+  private fun matchByIds(searchCriteria: PrisonerListCriteria): BoolQueryBuilder? {
+    with(searchCriteria) {
```

Comment:
looks like the `with` is redundant here, you tend to use in Kotlin so you don't have to `searchCriteria.prisonerNumbers` and use `prisonerNumbers`, given there is one statement, I would remove the `with`